contrib/format-daffodil/src/main/java/org/apache/drill/exec/store/daffodil/DaffodilDrillInfosetOutputter.java

```diff
@@ -0,0 +1,180 @@
+import org.apache.drill.exec.vector.accessor.writer.MapWriter;
+import org.apache.drill.exec.record.metadata.TupleMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Enumeration;
+
+import java.util.Stack;
+
+/**
+ * Adapts Daffodil parser infoset event calls to Drill writer calls
+ * to fill in Drill data rows.
+ */
+public class DaffodilDrillInfosetOutputter
+    extends BlobMethodsImpl implements InfosetOutputter {
+
+  private final Stack<ColumnWriter> columnWriterStack = new Stack<ColumnWriter>();
+
+  private ColumnWriter columnWriter() {
+    return columnWriterStack.peek();
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(DaffodilDrillInfosetOutputter.class);
+
+  private DaffodilDrillInfosetOutputter() {} // no default constructor
+
+  public DaffodilDrillInfosetOutputter(RowSetLoader writer) {
+    columnWriterStack.push(writer);
+  }
+
+  @Override
+  public void reset() {
+  }
+
+  @Override
+  public void startDocument() {}
+
+  @Override
+  public void endDocument() {}
+
+  private String colName(ElementMetadata erd) {
+    NamedQName nqn = erd.namedQName();
+    String colName = nqn.toPrefix_localName(); // just the local name if no prefix/namespace
+    return colName;
+  }
+
+  @Override
+  public void startSimple(InfosetSimpleElement diSimple) {
+    ElementMetadata erd = diSimple.metadata();
+    boolean isNilled = diSimple.isNilled();
+    String colName = colName(erd);
+    // If the column is an array, startArray will have setup columnWriter() to be the array writer.
+    // Otherwise, it's a TupleMetadata and we need to get this simple element's column.
+    // or it could be a top-level simple element (which we do use for test cases)
+    // in which case the columnWriter is the original RowSetWriter.
+    System.err.println(columnWriter().getClass().getName());
+    ColumnWriter cw = columnWriter();
+    switch (cw.type()) {
+      case ARRAY: {
+        assert(erd.isArray());
+        // do nothing startArray has this ready to write.'
+        break;
+      }
+      case TUPLE: {
+        cw = ((TupleWriter)cw).column(colName);
+        break;
+      }
+    }
+    ColumnMetadata cm = cw.schema();
+    if (isNilled) {
+      assert(cm.isNullable());
+      cw.setNull(); // Can I have a nullable array item. I.e., each item is nullable?
+    } else {
+      //
+      // FIXME: only INT is implemented right now.
+      //
+      int value = diSimple.getInt(); // will fail on downcast if not an INT.
+      cw.setObject(value); // autoboxing for primitive types.
+    }
+  }
+
+  @Override
+  public void endSimple(InfosetSimpleElement diSimple) {
+     /// nothing to do.
+  }
+
+  @Override
+  public void startComplex(InfosetComplexElement diComplex) {
+    ComplexElementMetadata erd = diComplex.metadata();
+    String colName = colName(erd);
+    ColumnWriter cw = columnWriter();
+    switch (cw.type()) {
+      case ARRAY: {
+        //
```

Comment:
This issue is fixed, at least as far as a test reads 3 records each containing two fields.